src/pybind/mgr/volumes/fs/health.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import threading
+
+from .exception import VolumeException
+
+log = logging.getLogger(__name__)
+
+# singleton design pattern taken from http://www.aleax.it/5ep.html
+
+class HealthUpdateSingleton(object):
+    """
+    Singleton class to track health metadata for various "modules"
+    (and send update to `ceph status`). Severity of health is left
+    upto the module.
+
+    This is implemented as a singleton to "funnel" health updates
+    for a given type (else, this can be instantiated thousand times
```

Comment:
'type' of what? Can you please elaborate?

Reply:
`type` here is an arbitary key that's used to identify a "health module" in this class.